detector.go

```diff
@@ -244,7 +245,7 @@ func (d *Detector) detectGroup(group buildpack.Group, done []buildpack.GroupElem
 				} else {
 					for i := range bpDescriptor.Targets {
 						d.Logger.Debugf("Checking for match against descriptor:", bpDescriptor.Targets[i])
-						if d.AnalyzeMD.RunImage.TargetMetadata.IsSatisfiedBy(&bpDescriptor.Targets[i]) {
+						if platform.TargetSatisfiedForBuild(*d.AnalyzeMD.RunImage.TargetMetadata, bpDescriptor.Targets[i]) {
```

Comment:
mixed feelings on turning this from "function with a receiver and one arg" into "function with two args" but I think you got the name right

Reply:
Yeah... the former signature was so much more readable. But then, it would have to live in the `files` package 😞